adobe/copy_on_write.hpp

```diff
@@ -192,7 +182,7 @@ class copy_on_write {
 
     \return A const reference to the underlying object
     */
-    const value_type& operator*() const { return read(); }
+    auto operator*() const noexcept -> const element_type& { return read(); }
```

Comment:
Is the return type expression necessary here? It seems to me the deduced type from `read()` will suffice.